fili-core/src/main/java/com/yahoo/bard/webservice/util/EnumUtils.java

```diff
@@ -9,6 +9,27 @@
  * Helper methods for enums.
  */
 public class EnumUtils {
+    /**
+     * Boilerplate method to retrieve an enum method via an arbitrary key, rather than just the enum value.
+     *
+     * @param <T>  Enum we are getting the value by key for
+     * @param <K>  Type of the key to get the enum value for
+     * @param key  Key to get the enum value for
+     * @param mapping  Mapping to use to look up the value by key
+     * @param enumeration  Enum we are getting value by key for (needed due to type erasure)
+     *
+     * @return The enum value for the key
+     *
+     * @throws IllegalArgumentException if this enum type has no constant with the specified name
+     */
+    public static <T extends Enum<T>, K> T forKey(K key, Map<K, T> mapping, Class<T> enumeration) {
```

Comment:
I'd love to see a test for more than just `String` as the key type. Perhaps I missed it, but I didn't see any tests that cover the non-String case.